### PR TITLE
[X] do not register srcinfo for xaml-defined values

### LINKED ISSUE #17222: VisualDiagnostics.GetSourceInfo on Style object returns the wrong source location.
### Description

One of the core strength of the  Live Property Explorer (LPE) is to not only display the list of the properties currently set on a particular target object, but also to be able to see the cascading style (the chain) of the style that contribute to the property value.  In other words, each property value in the LPE is sorted by the Value Source, we need to be able to locate the source of a Style Object. 

### Steps to Reproduce

Download [PropertyValueChain.zip](https://github.com/dotnet/maui/files/12528204/PropertyValueChain.zip)

Open the PropertyValueChain/MAUI/MAUI.sln.   

Run the App 
Put a break point into projects/PropertyValueChain/MAUI/MainPage.xaml.cs 'MainButtonClicked'

Read the comment: 
        // Requirement #3 Getting SourceInfo for Style Object
            info = VisualDiagnostics.GetSourceInfo(style);
            Debug.WriteLine($"Style Name:{style.BaseResourceKey} SourceUri:{info.SourceUri}, Line:{info.LineNumber}, Column:{info.LinePosition}");
            // Output: Error!!!, Wrong Source URI Style Name: SourceUri:MainPage.xaml;assembly=MAUI, Line:18, Column:17

To support Property Value Source, it is required that the Style object's source info return the source location of where the style is defined, not where it's been used in the page. 

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-preview.7.8842

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change

For values defined in other XAML file, the source info of the actual source is the correct one. Keep th eold path for values defined in XAML

### Issues Fixed

- fixes #17222

### CODE REVIEW

File: src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			//		Debug.WriteLine($"Setter Property:{setter.Property.PropertyName}, Setter Value:{setter.Value.ToString()}");
+
+			//		// Output: OK! Setter Property:TextColor, Setter Value:[Color: Red=1, Green=0, Blue=0, Alpha=1] (MainStyle.xaml)
+			//		// Output: OK! Setter Property:WidthRequest, Setter Value:100 (BaseStyle.xaml)
+			//		// Output: OK! Setter Property:HeightRequest, Setter Value:100 (BaseStyle.xaml)
+			//		// Output: OK! Setter Property:TextColor, Setter Value:[Color: Red=0, Green=0, Blue=1, Alpha=1]
+			//	}
+
+			//	// Requirement #4, Walking the Style Parent Chain
+			//	style = style.BasedOn;
```
Comment (jfversluis) [inline src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs:91]: Do we still need this?

**Review by jfversluis — DISMISSED** [review]

**Review by samhouts — DISMISSED** [review]

**Review by chabiss — DISMISSED** [review]

**Review by jfversluis — DISMISSED** [review]

File: src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	{
+		//this stub will be replaced at compile time
+	}
+
+	[TestFixture]
+	class Test
+	{
+		[SetUp] public void Setup() => AppInfo.SetCurrent(new MockAppInfo());
+		[TearDown] public void TearDown() => AppInfo.SetCurrent(null);
+
```
Comment (StephaneDelcroix) [inline src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs:29]:
```suggestion
#if DEBUG
```

File: src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			Assert.AreEqual(6, info.LineNumber);
+			Assert.AreEqual(10, info.LinePosition);
+
+
+			style = style.BasedOn;
+			info = VisualDiagnostics.GetSourceInfo(style);
+			Assert.AreEqual(new Uri("Issues/Maui17222BaseStyle.xaml;assembly=Microsoft.Maui.Controls.Xaml.UnitTests", UriKind.Relative), info.SourceUri);
+			Assert.AreEqual(5, info.LineNumber);
+			Assert.AreEqual(6, info.LinePosition);
+		}
```
Comment (StephaneDelcroix) [inline src/Controls/tests/Xaml.UnitTests/Issues/Maui17222.xaml.cs:70]:
```suggestion
#ENDIF
		}
```

**Review by jfversluis — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Pull request merged**